### PR TITLE
[BugFix] Fix kafka message body length is 0 for avro deserialization

### PR DESCRIPTION
**Pull request by zaorangyang**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

File: be/src/runtime/routine_load/data_consumer_group.cpp [hunk 1]
```diff
@@ -239,22 +239,25 @@ Status KafkaDataConsumerGroup::start_all(StreamLoadContext* ctx) {
...
-                    if (err) {
-                        LOG(ERROR) << "serdes deserialize avro failed: " << errstr;
-                        return Status::InternalError("serdes deserialize avro failed");
-                    }
-                    char* as_json;
-                    if (avro_value_to_json(&avro, 1, &as_json)) {
-                        LOG(ERROR) << "avro to json failed: %s" << avro_strerror();
-                        return Status::InternalError("avro to json failed");
+                    // We must ensure the msg len > 0.
+                    if (msg->len() > 0) {
```
Comment (howrocks) [inline be/src/runtime/routine_load/data_consumer_group.cpp:243]: Nothing need to do in `else` branch?

**Review by mark-phoenix-ai — DISMISSED** [review]

File: be/src/runtime/routine_load/data_consumer_group.cpp (hunk 1 above)
Comment (zaorangyang) [reply be/src/runtime/routine_load/data_consumer_group.cpp]: Yes, we keep our old behavior.

File: be/src/runtime/routine_load/data_consumer_group.cpp [hunk 2]
```diff
@@ -239,22 +239,25 @@ Status KafkaDataConsumerGroup::start_all(StreamLoadContext* ctx) {
...
+                        avro_value_t avro;
+                        DeferOp op([&] { avro_value_decref(&avro); });
+                        serdes_schema_t* schema;
+                        serdes_err_t err = serdes_deserialize_avro(serdes, &avro, &schema, msg->payload(), msg->len(),
+                                                                   errstr, sizeof(errstr));
+                        if (err) {
+                            LOG(ERROR) << "serdes deserialize avro failed: " << errstr;
+                            return Status::InternalError("serdes deserialize avro failed");
```
Comment (meegoo) [inline be/src/runtime/routine_load/data_consumer_group.cpp:251]: Return specific error messages as much as possible, we may not be able to see the log in some case such as SAAS.

File: be/src/runtime/routine_load/data_consumer_group.cpp [hunk 3]
```diff
@@ -239,22 +239,25 @@ Status KafkaDataConsumerGroup::start_all(StreamLoadContext* ctx) {
             } else {
                 Status st = Status::OK();
                 if (ctx->format == TFileFormatType::FORMAT_AVRO) {
```
Comment (chaoyli) [inline be/src/runtime/routine_load/data_consumer_group.cpp:241]:
```suggestion
                if (ctx->format == TFileFormatType::FORMAT_AVRO && msg->len() > 0) {
```

Comment (zaorangyang) [reply be/src/runtime/routine_load/data_consumer_group.cpp:241]: > 

To do so would enter else logic, which is not right
image

File: be/src/runtime/routine_load/data_consumer_group.cpp (hunk 2 above)
Comment (zaorangyang) [reply be/src/runtime/routine_load/data_consumer_group.cpp]: > Return specific error messages as much as possible, we may not be able to see the log in some case such as SAAS.

Updated

**Review by chaoyli — APPROVED** [review]

**Review by meegoo — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by zaorangyang** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Pull request merged**